### PR TITLE
Use the client section of the community roadmap, since it gets more frequent updates

### PR DESCRIPTION
**Pull request by steiza**
#### Summary

As [discussed in the Sigstore Clients SIG meeting](https://docs.google.com/document/d/1PNbBZSG3QC8hWVYBx6YDppaXwmSLDfx7t66ECaGa8y4/edit#heading=h.s3prc4a437fi), we want the roadmap to live in one place, and the community roadmap will be receiving updates more frequently.

#### Release Note

NONE

#### Documentation

N/A

### CODE REVIEW

**Review by loosebazooka — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -54,54 +54,7 @@ This Sig is co-chaired by [Fredrik Skogman]([url](https://github.com/kommendorka
 
 ## Roadmap
 
-You'll need to join [sigstore-dev@googlegroups.com](https://groups.google.com/g/sigstore-dev) for access to many of these (to prevent spam).
```
Comment (loosebazooka) [inline README.md:57]: should we leave this in in some capacity?

Comment (loosebazooka) [reply README.md:57]: nvm, its at the top.

**Review by woodruffw — APPROVED** [review]

**Comment by loosebazooka** [comment]
@steiza do you have permissions to merge?

**Comment by steiza** [comment]
> @steiza do you have permissions to merge?

I do not, but I think this is ready to land whenever someone who does have permissions is available!

**Pull request merged**